### PR TITLE
Switch to stream-based buffering for WARC export

### PR DESCRIPTION
**Pull request by tokee**
`StreamingSolrWarcExportBufferedInputStream` now holds a buffer of `InputStream`s instead of `byte[]` and uses a buffer for its `read(b, off, len)`-method, which should be faster.

There is no unit-test, but @thomasegense has a local test that he can run.

### CODE REVIEW

File: src/main/java/dk/kb/netarchivesuite/solrwayback/parsers/ArcParser.java [hunk 1]
```diff
@@ -235,7 +237,7 @@ private static void loadArcHeaderNotZipped(RandomAccessFile raf, ArcEntry arcEnt
   }
   
 
-  public static BufferedInputStream lazyLoadBinary(String arcFilePath, long arcEntryPosition) throws Exception{
+  public static BufferedInputStream lazyLoadBinary(String arcFilePath, long arcEntryPosition, long binarySize) throws Exception{
```
Comment (thomasegense) [inline src/main/java/dk/kb/netarchivesuite/solrwayback/parsers/ArcParser.java:239]: Dont think it is necessary to add binarySize to the API.
Both the warc and arc parser re-parse the header to the get file-pointer to the correct position. And in this 
parsing there binarySize is calculated again.

**Comment by thomasegense** [comment]
Only single a review comment. Fix if you agree.
Looked at the new  StreamingSolrWarcExportBufferedInputStream but take your words for it is ok :)

File: src/main/java/dk/kb/netarchivesuite/solrwayback/parsers/WarcParser.java [hunk 2]
```diff
@@ -253,7 +254,7 @@ public static ArcEntry getWarcEntryZipped(String warcFilePath, long warcEntryPos
   }
 
     
-    public static BufferedInputStream lazyLoadBinary(String arcFilePath, long arcEntryPosition) throws Exception{
+    public static BufferedInputStream lazyLoadBinary(String arcFilePath, long arcEntryPosition, long binarySize) throws Exception{
```
Comment (thomasegense) [inline src/main/java/dk/kb/netarchivesuite/solrwayback/parsers/WarcParser.java:257]: Dont think it is necessary to add binarySize to the API.
Both the warc and arc parser re-parse the header to the get file-pointer to the correct position. And in this
parsing there binarySize is calculated again.
Sp same issue in lazyloadBinary method in ArcParser.

**Review by thomasegense — CHANGES_REQUESTED** [review]

File: src/main/java/dk/kb/netarchivesuite/solrwayback/parsers/ArcParser.java (hunk 1 above)
Comment (tokee) [reply src/main/java/dk/kb/netarchivesuite/solrwayback/parsers/ArcParser.java]: Nice catch. Fixed.

**Pull request merged**